docs/contributor_guide/CONTRIBUTING.md

```diff
@@ -63,6 +63,10 @@ requests must be reviewed by at least one member of the team before merging.
 Reviews should follow the [pull request template](pr-template). If we want review on code that does not yet meet the definition of done, we open a draft
 pull request. Once a branch has been reviewed, it can be merged. We prefer to use squash merges, in order to simplify the `main` branch commit history. After merging the feature branch should be deleted.
 
+### Functions
+
+We prefer writing functions over classes to make it easier for beginners to understand the code. [Type hints](https://docs.python.org/3/library/typing.html) should be used when writing functions. We prefer functions to return `pandas.DataFrame` rather than `pandas.Series`. 
```

Comment:
Markdown sections says that we reference external links at the bottom of the Markdown file for clarity. This is the govcookiecutter default guidance so we don't have to stick to it, but for consistency we should either follow the guidance or remove it. 
Also consider adding context over when we prefer dataframes over series, for example when deriving a new (temporary) variable